### PR TITLE
practice pull request

### PR DESCRIPTION
**Pull request by annika-polavarapu**

### CODE REVIEW

File: practice_project/__pycache__/isPrime.cpython-311.pyc
Comment (heemonsu) [inline practice_project/__pycache__/isPrime.cpython-311.pyc:1]: Typically, we don't want to add files that are part of the build artifacts, and only want to keep the source code files around. Git provides a way to ignore sets of files, and usually `*.pyc` files are a good candidate for that.

I would highly recommend you set that up. You will just need to create a file named `.gitignore` in the root directory of your repo/project(this is a special filename that git recognises) and then commit that file in your repo. This will make it so that any new `*.pyc` in your project are ignored automatically.

For the file you have already commited, to remove them follow the steps here: https://stackoverflow.com/a/7528016

Let me know if you need any help here!

File: practice_project/__pycache__/test_isPrime.cpython-311-pytest-7.4.0.pyc
Comment (heemonsu) [inline practice_project/__pycache__/test_isPrime.cpython-311-pytest-7.4.0.pyc:1]: Please remove and ignore all `*.pyc` files.

File: practice_project/test_isPrime.py
Comment (heemonsu) [inline practice_project/test_isPrime.py:1]: These tests make sense. However, all of the asserts here essentially check the same thing. 

What happens if we add tests for some more 'edge' cases. I like the tests to handle negative numbers here. What if we have a `0`? What if someone gives in a number that's not an integer?

File: practice_project/isPrime.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        return True
+    
+    i = 2
+    while i * i <= number: # check the divisibility of each integer up to the square root of the number
+         if number % i == 0:
+              return False
+         i += 1
+    return True
+         
+if __name__ == '__main__':
```
Comment (heemonsu) [inline practice_project/isPrime.py]: nit: Since you have a test file that basically does the same thing, I would just remove the `main` block here!

File: practice_project/oddOrEven.py [hunk 2]
```diff
@@ -2,4 +2,8 @@ def is_odd_or_even(number):
     if number % 2 == 0:
         return "Even"
     else:
-        return "Odd"
\ No newline at end of file
+        return "Odd"
+    
+if __name__ == '__main__':
```
Comment (heemonsu) [inline practice_project/oddOrEven.py]: nit: Remove `main` block1

**Review by heemonsu — CHANGES_REQUESTED** [review]
Your PR looks good! I would just handle the gitignore for the `*.pyc` files before merging. The code comment are honestly just minor nits!

File: practice_project/isPrime.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+def is_prime(number) :
```
Comment (leonstaubach) [inline practice_project/isPrime.py]: I would recommend adding type hints for each method parameter and the return type of the method so that your IDE highlights misuse of the method (i.e. when you pass in a wrong type or you expect a wrong type of what is being returned

File: practice_project/test_isPrime.py
Comment (leonstaubach) [reply practice_project/test_isPrime.py:1]: I agree with the above. Furthermore, i would recommend breaking it down into multiple separate test methods, instead of grouping all asserts into one method, so that you get more precise test coverage insights (right now it would always be either 1/1 or 0/1 tests succeeded, but if you for example break it down into one method for positive integers, negative integer, misc type, etc. you'll have more precise test insights.
 
 As a general rule of thumb: each test method should usually only test "one thing" (one thing being defined as one specific case of code execution)

**Review by leonstaubach — COMMENTED** [review]
overall good! in addition to @heemonsu 's feedback i added some quality of life suggestions
